### PR TITLE
New unpacking progress dialog

### PR DESCRIPTION
**Pull request by marioba**
- Change unpacking message after 60%
- Use a custom dialog with a ProgressBar instead of the deprecated ProgressDialog class
- Use sensor orientation to launch Android's activity

### CODE REVIEW

**Comment by marioba** [comment]
@mbernasocchi please set the actual messages

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-cc33c3d6319f07f544e286f6d82954a1b9dddcc9-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-cc33c3d6319f07f544e286f6d82954a1b9dddcc9-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-cc33c3d6319f07f544e286f6d82954a1b9dddcc9-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-cc33c3d6319f07f544e286f6d82954a1b9dddcc9-x86_64.apk)

**Comment by mbernasocchi** [comment]
@marioba thanks will do in some hours

**Comment by nirvn** [comment]
Screenshot?

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-e9b6153953a26f23f9a6fc2c615dd74523a81c30-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-e9b6153953a26f23f9a6fc2c615dd74523a81c30-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-e9b6153953a26f23f9a6fc2c615dd74523a81c30-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-e9b6153953a26f23f9a6fc2c615dd74523a81c30-x86_64.apk)

**Comment by marioba** [comment]
@nirvn yes, thanks. I'm still working on it. At the moment the strings are only placeholders and also the layout will change soon.

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-012c349bc4616c2cd430e832f537c6ec260a0ed0-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-012c349bc4616c2cd430e832f537c6ec260a0ed0-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-012c349bc4616c2cd430e832f537c6ec260a0ed0-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-012c349bc4616c2cd430e832f537c6ec260a0ed0-x86_64.apk)

**Comment by marioba** [comment]
Screenshot_20200103-172338
It looks like this at the moment. The texts are still to be fixed.

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-3084b8f2fcc9f7ca2415fc0af97dbcf035020d49-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-3084b8f2fcc9f7ca2415fc0af97dbcf035020d49-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-3084b8f2fcc9f7ca2415fc0af97dbcf035020d49-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-3084b8f2fcc9f7ca2415fc0af97dbcf035020d49-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-8144fcc5fef8fb96289f7b42cd892e7aa3a155ac-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-8144fcc5fef8fb96289f7b42cd892e7aa3a155ac-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-8144fcc5fef8fb96289f7b42cd892e7aa3a155ac-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-8144fcc5fef8fb96289f7b42cd892e7aa3a155ac-x86_64.apk)

**Comment by mbernasocchi** [comment]
Screenshot_20200103-181945_QField Beta 770
Screenshot_20200103-181948_QField Beta 770

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-67beb77653906512b633bf44bccb555c0304c355-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-67beb77653906512b633bf44bccb555c0304c355-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-67beb77653906512b633bf44bccb555c0304c355-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-67beb77653906512b633bf44bccb555c0304c355-x86_64.apk)

**Comment by mbernasocchi** [comment]
Screenshot_20200103-201146_QField Beta 770

**Comment by mbernasocchi** [comment]
@nirvn do we already have a background lighter green?

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-733d28a5ba39e7cfbecbf05fed468c3e6532e468-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-733d28a5ba39e7cfbecbf05fed468c3e6532e468-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-733d28a5ba39e7cfbecbf05fed468c3e6532e468-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-733d28a5ba39e7cfbecbf05fed468c3e6532e468-x86_64.apk)

**Comment by mbernasocchi** [comment]
Screenshot_20200103-213926_QField Beta 770
@nirvn feel free to change the backgroundTint

File: android/res/values/strings.xml [hunk 1]
```diff
@@ -3,7 +3,9 @@
     <string name="app_name" translatable="false">QField</string>
     <string name="lib_name" translatable="false">qfield</string>
     <string name="done_unpacking_msg">Done unpacking, start QGIS.</string>
-    <string name="unpacking_msg">Unpacking post install data. This might take some time.</string>
+    <string name="unpacking_title">QField is finishing its installation</string>
```
Comment (nyalldawson) [inline android/res/values/strings.xml]: "Finalizing install" has the same meaning, but is much less wordy!

File: android/res/values/strings.xml [hunk 2]
```diff
@@ -3,7 +3,9 @@
...
+    <string name="unpacking_msg">this one-time operation will take a brief moment</string>
```
Comment (nyalldawson) [inline android/res/values/strings.xml]: "This" (to match capitalisation of "Nearly there" message)

**Comment by ElvisDahnert** [comment]
Excuse me, I'm curious.
Why is unpacking done with landscape display only? This is true even without automatic screen rotation.

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-3d3e9dd2349b6547fb9997f7bae7b4603a6ab4d1-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-3d3e9dd2349b6547fb9997f7bae7b4603a6ab4d1-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-3d3e9dd2349b6547fb9997f7bae7b4603a6ab4d1-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-3d3e9dd2349b6547fb9997f7bae7b4603a6ab4d1-x86_64.apk)

**Comment by marioba** [comment]
@ElvisDahnert good question. 
I think it's only for historical reasons. I've modified it to use the sensor orientation of the device.

@mbernasocchi do you agree?

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-3e2a63a6f371e7189202533cac705b0dafb15535-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-3e2a63a6f371e7189202533cac705b0dafb15535-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-3e2a63a6f371e7189202533cac705b0dafb15535-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-3e2a63a6f371e7189202533cac705b0dafb15535-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-ee3c9ac1a684e2f0deae040a42de55f6ce5a4fcc-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-ee3c9ac1a684e2f0deae040a42de55f6ce5a4fcc-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-ee3c9ac1a684e2f0deae040a42de55f6ce5a4fcc-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-ee3c9ac1a684e2f0deae040a42de55f6ce5a4fcc-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-abfafb2f693f9249c1cdf44c592787a00c49e244-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-abfafb2f693f9249c1cdf44c592787a00c49e244-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-abfafb2f693f9249c1cdf44c592787a00c49e244-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-abfafb2f693f9249c1cdf44c592787a00c49e244-x86_64.apk)

File: android/res/values/strings.xml (hunk 1 above)
Comment (mbernasocchi) [reply android/res/values/strings.xml]: @nyalldawson it was a placeholder, I was thinking of: "QField is getting ready for work"
what do you think?

**Comment by mbernasocchi** [comment]
> @ElvisDahnert good question.
> I think it's only for historical reasons. I've modified it to use the sensor orientation of the device.
> 
> @mbernasocchi do you agree?

not sure, there was a reason, IIRC it used to break the unzipping if one would turn the screen. but i'm talking 2014 :) so I agree and let's test changing orientation while unpacking

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-f3a07899b0b16728ac1ef1c74d470a82b6166309-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-f3a07899b0b16728ac1ef1c74d470a82b6166309-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-f3a07899b0b16728ac1ef1c74d470a82b6166309-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-f3a07899b0b16728ac1ef1c74d470a82b6166309-x86_64.apk)

**Comment by ElvisDahnert** [comment]
> > @ElvisDahnert good question.
> > I think it's only for historical reasons. I've modified it to use the sensor orientation of the device.
> > @mbernasocchi do you agree?
> 
> not sure, there was a reason, IIRC it used to break the unzipping if one would turn the screen. but i'm talking 2014 :) so I agree and let's test changing orientation while unpacking

It is true. When the screen is rotated unpacking is restarted, only after holding the screen in one position is the unpacking completed.

**Comment by mbernasocchi** [comment]
> It is true. When the screen is rotated unpacking is restarted, only after holding the screen in one position is the unpacking completed.

@marioba please revert the orientation part

**Comment by marioba** [comment]
I'll try to fix it first. Or to disable the rotation during the unpacking.

**Comment by mbernasocchi** [comment]
> I'll try to fix it first. Or to disable the rotation during the unpacking.

ok, but don't lose too much time. @m-kuhn was wondering about getting rid of unpacking at all.

**Comment by marioba** [comment]
Ok, now the activity starts with the current rotation but it's not possible to change it during the unpacking.

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-84b77c8dd793adf02e974ce66d392295a45a2d97-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-84b77c8dd793adf02e974ce66d392295a45a2d97-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-84b77c8dd793adf02e974ce66d392295a45a2d97-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-84b77c8dd793adf02e974ce66d392295a45a2d97-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-b7001dcf6e2549fa3ec38e20e17f177f0518a51d-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-b7001dcf6e2549fa3ec38e20e17f177f0518a51d-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-b7001dcf6e2549fa3ec38e20e17f177f0518a51d-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-770-b7001dcf6e2549fa3ec38e20e17f177f0518a51d-x86_64.apk)

**Comment by mbernasocchi** [comment]
Screenshot_20200104-182141_QField Beta 770
Screenshot_20200104-182138_QField Beta 770

**Comment by mbernasocchi** [comment]
Screenshot_2020-01-04-20-32-17
Screenshot_2020-01-04-20-32-24

@marioba the rotation block works thanks. for me it's good to merge

**Comment by mbernasocchi** [comment]
@marioba the only issue on the tablet is that the logo looks very pixelated

**Comment by marioba** [comment]
@mbernasocchi This depends on the resolution of the logos in the `android/res/drawable-*` directories. It is probably worth checking them and possibly changing them to higher resolution versions. 
But this is a separate issue from this pull request.

File: android/src/ch/opengis/qfield/QFieldActivity.java [hunk 3]
```diff
@@ -45,285 +43,305 @@
...
+                makeSymlink(storagePath, storagePathAlias);
+            }
+        }
+
+        protected void onProgressUpdate(Integer... progress) {
+            super.onProgressUpdate(progress);
+            progressBar.setProgress(progress[0]);
+
+            if (progress[0] >= 60) {
+                unpackingMessage.setText(getString(R.string.unpacking_msg_following));
```
Comment (3nids) [inline android/src/ch/opengis/qfield/QFieldActivity.java]: the text will be updated each time, we don't care?

Comment (mbernasocchi) [reply android/src/ch/opengis/qfield/QFieldActivity.java]: I don't think it's an issue.

**Pull request merged**